sotodlib/hwp/g3thwp.py

```diff
@@ -790,21 +790,20 @@ def _calc_angle_linear(self):
             direction = self._force_quad
 
         self._encd_cnt_split = np.split(self._encd_cnt, self._ref_indexes)
-        self._angle = (self._encd_cnt_split[0] - self._ref_cnt[0]) * \
+        angle_first_revolution = (self._encd_cnt_split[0] - self._ref_cnt[0]) * \
             (2 * np.pi / self._num_edges) % (2 * np.pi)
-        self._angle = np.append(self._angle,
-                                np.concatenate(
-                                    np.array(
-                                        [((self._encd_cnt_split[i] - self._ref_cnt[i]) *
-                                          (2 * np.pi /
-                                            np.diff(self._ref_indexes)[i - 1])
-                                            % (2 * np.pi)).flatten()
-                                            for i in range(1, len(self._encd_cnt_split) - 1)], dtype=object)))
-        self._angle = np.append(self._angle,
-                                (self._encd_cnt_split[-1] - self._ref_cnt[-1]) *
-                                (2 * np.pi / self._num_edges) % (2 * np.pi))
-        self._angle = direction * self._angle % (2 * np.pi)
-
+        angle_last_revolution = (self._encd_cnt_split[-1] - self._ref_cnt[-1]) * \
+            (2 * np.pi / self._num_edges) % (2 * np.pi) + len(self._ref_cnt) * 2 * np.pi
+        self._angle = np.array([((self._encd_cnt_split[i] - self._ref_cnt[i]) * \
+                          (2 * np.pi / np.diff(self._ref_indexes)[i - 1]) \
+                              % (2 * np.pi) + i * 2 * np.pi ).flatten() \
+                                  for i in range(1, len(self._encd_cnt_split) - 1)])
+        self._angle = np.concatenate([angle_first_revolution, self._angle.flatten(), angle_last_revolution])
+        self._angle = direction * self._angle
+        if mod2pi:
+            self._angle = self._angle % (2 * np.pi)
+
+        ii = np.where(np.diff(self._ref_indexes) != 1140)[0]
```

Comment:
This line was deleted in current master branch. I just forget to delete in debug process...

Reply:
Okay this is my fault I'll delete again.